### PR TITLE
Fix strict nullable for default factory

### PR DESCRIPTION
**Pull request by koxudaxi**
Related Issues: https://github.com/koxudaxi/datamodel-code-generator/issues/1216

### CODE REVIEW

**Comment by tommyjcarpenter** [comment]
THANK YOU! Can this be released ?

**Comment by tommyjcarpenter** [comment]
Screenshot 2023-04-17 at 6 18 45 AM

**Comment by tommyjcarpenter** [comment]
Hi - I tried using this, but I am still getting the `Optional[List[str]]`, am I using this wrong?

File: tests/data/expected/main/main_use_default_kwarg/output.py [hunk 1]
```diff
@@ -72,3 +72,7 @@ class Name(BaseModel):
 
 class Tag(BaseModel):
     __root__: str
+
+
+class Notes(BaseModel):
+    comments: Optional[List[str]] = Field(default_factory=list)
```
Comment (tommyjcarpenter) [inline tests/data/expected/main/main_use_default_kwarg/output.py:78]: this should not have an optional.

Comment (koxudaxi) [reply tests/data/expected/main/main_use_default_kwarg/output.py:78]: @tommyjcarpenter 
I expect you to use `--strict-nullable` option for the behavior.
https://github.com/koxudaxi/datamodel-code-generator/blob/03eeddbabf9259dd083a4055809ba62d26178e6c/tests/test_main.py#L2641-L2665

Comment (koxudaxi) [reply tests/data/expected/main/main_use_default_kwarg/output.py:78]: But, if the object has `default_factory`, may remove `Optional[]` for the behavior :thinking:

Comment (tommyjcarpenter) [reply tests/data/expected/main/main_use_default_kwarg/output.py:78]: @koxudaxi ok thanks, trying now! I dont understand why you would ever have Optional if theres a default factory though

Comment (koxudaxi) [reply tests/data/expected/main/main_use_default_kwarg/output.py:78]: @tommyjcarpenter 
The behavior respects the type of object. But your thought makes sense.
I will fix it. Thank you

Comment (tommyjcarpenter) [reply tests/data/expected/main/main_use_default_kwarg/output.py:78]: @koxudaxi i confirmed this works using that flag, thank you!

**Pull request merged**